ValidationTest/Main.cs

```diff
@@ -60,20 +61,80 @@ public ValidationTest(Config config)
             }
         }
 
-        public static Histogram RequestDuration { get; } = Metrics.CreateHistogram(
-            "LodeRunnerDuration",
-            "Histogram of LodeRunner request duration",
-            new HistogramConfiguration
-            {
-                Buckets = Histogram.ExponentialBuckets(1, 2, 10),
-                LabelNames = new string[] { "category" },
-            });
-
         /// <summary>
         /// Gets UtcNow as an ISO formatted date string
         /// </summary>
         public static string Now => DateTime.UtcNow.ToString("o", CultureInfo.InvariantCulture);
 
+        public Histogram RequestDuration
+        {
+            get
+            {
+                if (config.Prometheus && requestDuration == null)
+                {
+                    requestDuration = Metrics.CreateHistogram(
+                    "LodeRunnerDuration",
+                    "Histogram of LodeRunner request duration",
+                    new HistogramConfiguration
+                    {
+                        Buckets = Histogram.ExponentialBuckets(1, 2, 10),
+                        LabelNames = new string[] { "code", "category", "mode", "server", "failed", "zone", "region" },
+                    });
+                }
+
+                return requestDuration;
+            }
+        }
+
+        public static string GetMode(PerfLog perfLog)
```

Comment:
It doesn't look like `perfLog.Category` is needed to determine the value of `mode`. I recommend moving that part out to something like `UpdateCategory(PerfLog perfLog)`.

Reply:
some categories aren't covered by the if statement, so it is needed.

we could modify it to return a tuple or an out param which is probably a better approach